sktime/forecasting/trend/_curve_fit_forecaster.py

```diff
@@ -0,0 +1,177 @@
+# !/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements FunctionForecaster."""
+
+__author__ = ["benheid"]
+__all__ = ["CurveFitForecaster"]
+
+import numpy as np
+import pandas as pd
+from scipy.optimize import curve_fit
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+
+class CurveFitForecaster(BaseForecaster):
+    """The CurveFitForecaster takes a function and tits it by using scipy curve_fit.
+
+    The CurveFitForecaster applies the scipy curve_fit method to find the best
+    parameter of a function. This function maps a list of integers to real values.
+    The integers are derrived from the index of the input time series.
+
+    In `fit`
+    1. The index of the input time series is transformed to a list of Integers.
+    2. The scipy curve_fit is called using the list of integers as x values,
+       and the time series values as y values.
+    In `predict`
+    1. The ForecastingHorizon is transformed to a list of integers.
+    2. The list of integers is passed together with the fitted parameters to the
+       function to provide the forecast.
+
+
+    Parameters
+    ----------
+    function: Callable
+        The function that should be fitted and used to make forecasts.
+        The signature of the functions is `function(x, ...)`.
```

Comment:
type is crucial, what is the type of the function?

Reply:
I added the input and return information.

Reply:
ok, thanks